src/library_html5.js

```diff
@@ -2759,6 +2759,28 @@ var LibraryJSEvents = {
     return !GL.contexts[target] || GL.contexts[target].GLctx.isContextLost(); // No context ~> lost context.
   },
 
+#if USE_WEBGPU
+  emscripten_webgpu_get_device__sig: 'i',
+  emscripten_webgpu_get_device: 'emscripten_webgpu_do_get_device',
+  emscripten_webgpu_do_get_device__deps: ['$WebGPU'],
+  emscripten_webgpu_do_get_device: function() {
+    // TODO(kainino0x): make it possible to actually create devices
+    assert(Module['preinitializedWebGPUDevice']);
+    WebGPU.initManagers();
```

Comment:
This `WebGPU.initManagers()` looks like a function that needs to be called just once to initialize the JS "system" state, and is not device instance/context instance specific. Therefore `WebGPU.initManagers()` probably does not need to be called on a second call to `emscripten_webgpu_get_device()`? Probably a `emscripten_webgpu_get_device__postset: 'WebGPU.initManagers();'` might be able to do the job.